### DISCUSSION TITLE
How to add another argument after -e option in qterminal

### DISCUSSION THREAD

**Discussion by anneranch** [lxqt/lxqt#2306] [Q&A]
While the attached code works as expected processing "qterminal -e bluetoothctl", I cannot figure out how to add 
another argument to make "qterminal -e bluetoothctl help"; 
(This does not work - combining arguments - "bluetoothctl  help" - QProcess rejects arguments with spaces. )
Any help would be greatly appreciated.

#ifdef TERMINAL_DEBUG
                {
                           qDebug()<< Q_FUNC_INFO;
                           processTERMINAL = new QProcess();
                           QString exec = "qterminal";

                           //params = QStringList() << "qterminal " << command;
                           //params = QStringList() << "qterminal " << " -e" << " bluetoothctl "; //  command;
                           // no go QStringList params = QStringList() << "qterminal" << "-e" << "bluetoothctl" <<"help\n";
                           QStringList params = QStringList() << "qterminal" << "-e" << "bluetoothctl"; //  help";
                           processTERMINAL->start(exec, params,QIODevice::ReadWrite);      /// QIODevice::ReadWrite is default anyway
                           if(processTERMINAL->waitForStarted())
                {
                           qDebug() << " processTERMINAL->waitForStarted() OK";
                           qDebug() << processTERMINAL->readAllStandardError();
                           qDebug() << processTERMINAL->readAllStandardOutput();
                           processTERMINAL->open();
                           processTERMINAL->write("help\n");

                }
                           return;
                }
                #endif

**Comment by tsujan** [comment]
Supposing that you've compiled the latest git QTerminal correctly, there's no difference between `qterminal -e` and `xterm -e` in this regard: the Help terminal window is shown and exits immediately.

See https://github.com/lxqt/qterminal/issues/554#issuecomment-471252822

**Reply by anneranch** [reply]
Suggesting  the alternative  does not answer the question . Questioning the compilation of QTerminal does not answer the question either.  Please read the post and stick with the subject Thanks

**Reply by tsujan** [reply]
Either your question is about QTerminal -- in which case, it was answered -- or it's about Qt coding, in which case, it isn't related to LXQt.

You seem not to want to read, and by chance, `read` is the key:

```c++
    QString exec = "qterminal";
    QStringList params = QStringList()  <<"-e" << "bash -c 'bluetoothctl help  && read'";
    processTERMINAL->start(exec, params);
```

**Reply by yan12125** [reply]
> Questioning the compilation of QTerminal does not answer the question either.

Nope. tsujan is correct - you need to compile qterminal to get it work.

**Comment by anneranch** [comment]
Suggested solution did not solve the issue. QProcess starts and there is no
qterminal output window / dialog.
Spill your opinions in social media forums.

**Comment by anneranch** [comment]
Nope. tsujan is correct - you need to compile qterminal to get it work.

So is "you must compile from source " the answer ?
I hope NOT.

**Reply by yan12125** [reply]
If you don't want to compile the latest qterminal, you can create a temporary script and run it. For example, create a file `temp-script.sh`:
```
bluetoothctl help
read
```
and run `qterminal -e bash temp-script.sh`
